src/Okta.Sdk.UnitTests/UserFactorsClientShould.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NSubstitute;
+using Okta.Sdk.Internal;
+using Okta.Sdk.UnitTests.Internal;
+using Xunit;
+
+namespace Okta.Sdk.UnitTests
+{
+    public class UserFactorsClientShould
+    {
+        [Fact]
+        public async Task EnrollEmailFactor()
+        {
+            var mockDataStore = Substitute.For<IDataStore>();
+            var client = new TestableOktaClient(mockDataStore);
+            var factorsClient = client.UserFactors;
+            var emailFactorOptions = new AddEmailFactorOptions
+            {
+                Email = "johndoe@mail.com",
+                TokenLifetimeSeconds = 999,
+            };
+            await factorsClient.AddFactorAsync("UserId", emailFactorOptions);
+            await mockDataStore
+                .Received(1)
+                .PostAsync<UserFactor>(Arg.Is<HttpRequest>(request => RequestMatchOptions(request, emailFactorOptions)), Arg.Any<RequestContext>(), Arg.Any<CancellationToken>());
```

Comment:
For testing in general, we usually favor readability over reusability. The idea is that tests should be easy to read and follow by reviewers (especially when we add a new API because the reviewer has to take a look at a considerable number of tests, and we should facilitate that process). You can see this pattern in _almost_ all our tests. 

I suggest we inline this logic here instead of using a separate private method. Also, in addition to verifying that the query params are correct, we should check not only that the payload has all the fields we set but also the structure (correct nesting, etc.).